guppylang/std/qsystem/random.py

```diff
@@ -80,3 +86,16 @@ def _random_float(self: "RNG") -> float: ...
     @guppy.custom(RandomIntBoundedCompiler(), module=qsystem_random)
     @no_type_check
     def _random_int_bounded(self: "RNG", bound: int) -> int: ...
+
+    @guppy(qsystem_random)
+    @no_type_check
+    def shuffle(self: "RNG", array: array[SHUFFLE_T, SHUFFLE_N]) -> None:
+        """Randomly shuffle the elements of a possibly linear array in place.
+        Uses the Fisher-Yates algorithm."""
+        for k in range(SHUFFLE_N):
+            i = SHUFFLE_N - 1 - k
```

Comment:
```suggestion
        for i in range(SHUFFLE_N - 1, 0, -1):
```
Would it be more transparent to write the iteration this way?

Reply:
It would but sadly range doesn't work that way in guppy 

See #791